### ISSUE TITLE
Export `RE.Regex`, and make constructors easier.

### ISSUE THREAD

**Issue by panxiaoguang** [BioJulia/BioSequences.jl#167]
I want to have a regex search search for biosequences, and it can be changed by a value.

for example,

```julia

start="AGCAT"
re=@biore_str(string(start,"[AGCT]+"),"d")
```
it always repored an error, but I want it can be changed by start value?

**Comment by jakobnissen** [comment]
You can do this by explicitly constructing the regex:
```
julia> start="AGCAT"
"AGCAT"

julia> BioSequences.RE.Regex{DNA}(start * "[ACGT]+", :pcre)
biore"AGCAT[ACGT]+"dna
```
This functionality ought to be exposed - in general it's good style that, whenever you have a macro exported, also export the equivalent function.

**Comment by jakobnissen** [comment]
Resolved in #169
